retrievalmarket/impl/providerstates/provider_states_test.go

```diff
@@ -119,22 +120,6 @@ func TestReceiveDeal(t *testing.T) {
 		require.NotEmpty(t, dealState.Message)
 	})
 
-	t.Run("response write error", func(t *testing.T) {
```

Comment:
write responses are moved down to the decisioning function; this test is moved there as well.